include/picongpu/fields/MaxwellSolver/Substepping/Substepping.hpp

```diff
@@ -126,9 +129,31 @@ namespace picongpu
                     this->updateBeforeCurrent(static_cast<float_X>(currentStep));
                 }
 
+                /** Add contribution of FieldJ in the given area according to Ampere's law
+                 *
+                 * @tparam T_area area to operate on
+                 */
+                template<uint32_t T_area>
+                void addCurrent()
+                {
+                    // Since this is also called internally, check if currents exist in the simulation
+                    using namespace pmacc;
+                    using SpeciesWithCurrentSolver =
+                        typename pmacc::particles::traits::FilterByFlag<VectorAllSpecies, current<>>::type;
+                    constexpr auto numSpeciesWithCurrentSolver = bmpl::size<SpeciesWithCurrentSolver>::type::value;
```

Comment:
It was already wrong before but we need to check if a species has a current solver or if field background J is active.
Nothing for this PR, we will fix this in an additional PR.

Reply:
Yes, i just made an issue for it yesterday #4244